django/db/models/lookups.py

```diff
@@ -389,6 +389,19 @@ class LessThanOrEqual(FieldGetDbPrepValueMixin, BuiltinLookup):
     lookup_name = "lte"
 
 
+class IntegerFieldOverflow:
+    overflow_exception = EmptyResultSet
+
+    def process_rhs(self, compiler, connection):
+        rhs = self.rhs
+        if isinstance(rhs, int):
+            field_internal_type = self.lhs.output_field.get_internal_type()
+            _, max_value = connection.ops.integer_field_range(field_internal_type)
```

Comment:
We can also exceed the min value, e.g.
```diff
diff --git a/tests/model_fields/test_integerfield.py b/tests/model_fields/test_integerfield.py
index 0dbc82383a..7f8e630828 100644
--- a/tests/model_fields/test_integerfield.py
+++ b/tests/model_fields/test_integerfield.py
@@ -97,7 +97,7 @@ class IntegerFieldTests(TestCase):
             instance.full_clean()
 
     def test_backend_range_lookups(self):
-        max_value = self.backend_range[-1]
+        min_value, max_value = self.backend_range
         if max_value is None:
             raise SkipTest("Backend doesn't define an integer max value")
         overflow_value = max_value + 1
@@ -108,6 +108,8 @@ class IntegerFieldTests(TestCase):
             self.model.objects.get(value__gt=overflow_value)
         with self.assertNumQueries(0), self.assertRaises(self.model.DoesNotExist):
             self.model.objects.get(value__gte=overflow_value)
+        with self.assertNumQueries(0), self.assertRaises(self.model.DoesNotExist):
+            self.model.objects.get(value__lt=min_value - 1)
         with self.assertNumQueries(1):
             self.assertEqual(self.model.objects.get(value__lt=overflow_value), obj)
         with self.assertNumQueries(1):
```
I think we should check both.

Reply:
@felixxm happy to implement the below min value case as a separate commit as it's more of an optimization though

Reply:
[This](https://github.com/django/django/pull/16599#discussion_r1129334934) crashes on SQLite with `OverflowError`, so for me it's the same issue :thinking: Maybe I'm missing sth.

Reply:
@felixxm you are right, I think I missed it due to an off-by-one error when running checks locally. Both commits can be merged then I guess.